### PR TITLE
add media player client-side state

### LINKED ISSUE #191: Setup app wide media player state in app
## User Story

As an app user I want a seamless and integrated video and audio experience so that the app does not distract me from the content.

#### Problem Details _(Optional)_

* Screenshots, GIFs, or a video
* Any additional information (version, device, etc.)

#### Acceptance Criteria _(Optional)_

* [ ] It should [SPEC_DETAIL]

### PR DESCRIPTION
**Pull request by conrad-vanl**
Fixes #191 

## Creator

- [x] Build relevant tests if any apply
- [x] Ensure there are no new warnings (in app AND in storybook)
- [x] Upload an animated GIF showcasing the solution (if it applies)
- [x] Set a relevant reviewer
- [x] Add a line to the changelog documenting the high-level change this PR introduces

## Reviewer

- [ ] Review CI build results
- [ ] Review function and form on iOS
- [ ] Review function and form on Android
- [ ] Review new stories if any apply
- [ ] Review test coverage: Are all new or changed states tested?
- [ ] Review new test logic
- [ ] Review the changelog entry

### CODE REVIEW

File: mobile/src/store/index.js [hunk 1]
```diff
@@ -23,5 +51,78 @@ export const resolvers = {
...
+            nowPlayingIndex
+            queue
+          }
+        }
+      `;
+      const { mediaPlayer } = cache.readQuery({ query });
+      console.log({ mediaPlayer });
+
+      let nowPlayingIndex = mediaPlayer.nowPlayingIndex + skip;
+      if (nowPlayingIndex < 0) nowPlayingIndex = mediaPlayer.queue.length - 1;
```
Comment (vinnyjth) [inline mobile/src/store/index.js]: Should this logic be different if `shuffle` is set to true?

**Comment by vinnyjth** [comment]
@conrad-vanl This PR looks fantastic!! I think this is the first time I've seen `clientState` used in Apollo and I'm impressed already. I left a note about `shuffling` that is pretty minor, as soon as you let me know I can approve this :)

**Comment by conrad-vanl** [comment]
Good catch @vinnyjth, I think I'll opt to just remove some of these attributes all together for now, as we don't technically need them yet and they will be just as easy to implement later (I don't really want to spend time writing tests to prove that shuffling and repeating works and take on that code debt if we don't need to yet)

**Comment by vinnyjth** [comment]
@conrad-vanl Sounds great!

**Comment by conrad-vanl** [comment]
@vinnyjth done :)

**Pull request merged**